### PR TITLE
Delete A node in linked list CPP #94

### PR DESCRIPTION
**Pull request by ChetasShree**
## Related Issue
- Delete a Node in Linked List #94 

## Proposed Changes
- Added the cpp file of code
- added the readme file for the same

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [x] New feature (non-breaking change which adds functionality)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Image of readme file   |  Image of the code     |
| 

  | 

    |

### CODE REVIEW

**Review by Lakhankumawat — APPROVED** [review]
Code and README Looks Great , i know you @ChetasShree  from linked In but i did not expected you'll do the same thing as other contributors did , you checked the checklist but did not actually followed it .

**Comment by Lakhankumawat** [comment]
Any of mentors will review your pr shortly meanwhile please complete the checks .

File: L-LinkedList/Delete a node in LinkedList/delete_node_ll.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+If the node to be deleted is the head node, then simply point the head to the second node of the linked list.
+For all other nodes:
+Traverse the linked list and for the current node curr, check whether the next node contains the key that needs to be deleted.
+If true, point curr -> next to curr -> next -> next.
+Else, keep traversing the list until the end of the linked list.
+The following illustration will make the approach more clear.
+
+![ll](https://user-images.githubusercontent.com/75165587/156487339-0cbead17-c8c7-4ec2-8ee5-45120dcca7db.PNG)
+
+## CODE
```
Comment (n4i9kita) [inline L-LinkedList/Delete a node in LinkedList/delete_node_ll.md:28]: Please don't add code in Readme. 
- algorithm 
- pseudocode
can be added here.

**Review by miraj0507 — COMMENTED** [review]
kindly do a user input code of taking the inputs from user, so that it becomes more generalizes, rest looks good to me

File: L-LinkedList/Delete a node in LinkedList/delete_node_ll.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    new_node->next = (*head_ref);
+    (*head_ref) = new_node;
+}
+
+// function to delete a node in linked list
+Node *deleteNode(Node *head, int i)
+{
+    if (i == 1)
+        return head->next; // if we want to delete the head node then return the next node after head
+    Node *copyhead = head;
```
Comment (n4i9kita) [inline L-LinkedList/Delete a node in LinkedList/delete_node_ll.cpp:26]: Yep, `copyhead` pointer is initialized with head, but a better pointer name will be preferred.

File: L-LinkedList/Delete a node in LinkedList/delete_node_ll.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// cresting a node class for linked list
+class Node
+{
+public:
+    int data;
+    Node *next;
+};
+
+// function to pushing the elements into the linked list
+void push(Node **head_ref, int new_data)
```
[the remark this answers is not in the document]
Comment (miraj0507) [reply L-LinkedList/Delete a node in LinkedList/delete_node_ll.cpp:13]: yes agreed

**Comment by Lakhankumawat** [comment]
Incomplete or faulty checks , please review your PR checklists and make sure you follow them , thanks/.

**Comment by ChetasShree** [comment]
> Incomplete or faulty checks , please review your PR checklists and make sure you follow them, thanks/.

may I know what are the errors?

**Comment by Lakhankumawat** [comment]
> I.e. You haven't starred the repo but you checked it in your checklist 🤔

.. i don't if you guys go through comments or not

**Comment by Lakhankumawat** [comment]
> Any of mentors will review your pr shortly meanwhile please complete the checks .

..2nd time

**Comment by Lakhankumawat** [comment]
> Incomplete or faulty checks , please review your PR checklists and make sure you follow them , thanks/.

3rd time obviously

**Comment by Lakhankumawat** [comment]
Merging because your PR is good ... Please do not check the checklist blindfolded okay .

**Review by Lakhankumawat — APPROVED** [review]
LGTM

**Comment by Lakhankumawat** [comment]
@contributors-all please add @ChetasShree  code

**Comment by Lakhankumawat** [comment]
@all-contributors please add @ChetasShree  code ,

**Comment by Lakhankumawat** [comment]
@all-contributors please add @ChetasShree  code ,

**Pull request merged**